### PR TITLE
changed defi++ to prefer buy over mint

### PR DESCRIPTION
**Pull request by jordaniza**
I wanted to look at a clean way to add DEFI++ as an option to buy/sell in the main exchange modal.

Reviewing the codebase, the reason it does not appear is because the setting `preferMintOverBuy` is set to `true`

Changing this setting adds the token to the swap list on `Tokensswap.svelte` and `Exchange.svelte`, as we want. It also has the following effects on `Dashboard.svelte`:

- When opening up the list of products (Pies list on `Dashboard.svelte`), clicking the `buy` button links directly to the swap page, instead of to the DEFI++ page:

```svelte
                {#if pie.buyButton}
                  {#if pie.useMintOverBuy}
                      <a href={`#/pie/${pie.address}`}>
                        <button class="table-btn highlight-box min-w-70px">
                          {pie.symbol}
                        </button>
                      </a>
                  {:else}
                    <a href={`#/swap`}>
                      <button class="table-btn highlight-box min-w-70px">
                        {pie.symbol}
                      </button>
                    </a>
                  {/if}
                {/if}
```
```svelte
              {#if pie.useMintOverBuy}
                <button on:click={() => {
                  modalOption.token = pie.address;
                  modal.open();
                }} class="table-btn highlight-box min-w-70px">
                  {pie.symbol}
                </button>
              {:else}
                <a href={`#/swap`}>
                  <button class="table-btn highlight-box min-w-70px">
                    {pie.symbol}
                  </button>
                </a>
              {/if}
```
 ```svelte
                 {#if pie.buyButton}
                  {#if pie.useMintOverBuy}
                      <a href={pie.symbol == "SLICE" ? '#/slice' : `#/pie/${pie.address}`}>
                        <button class="table-btn highlight-box min-w-70px">
                          {pie.symbol}
                        </button>
                      </a>
                  {:else}
                    <a href={`#/swap`}>
                      <button class="table-btn highlight-box min-w-70px">
                        {pie.symbol}
                      </button>
                    </a>
                  {/if}
                {/if}
```

In summary: it appears to me to be absolutely fine to make this change, enabling the wider functionality

### CODE REVIEW

**Comment by Alexintosh** [comment]
Change DAI with USDC, check decimals

**Review by gbonumore — APPROVED** [review]

**Review by AdrianAddesso — CHANGES_REQUESTED** [review]
Almost there. One issue found.

**Fix needed**
❌  DEFI++ goes to exchange page <- It does, but the exchange does not Show DeFi++ listed when accessed from the PiePage

**Passed testing**
✅  PLAY goes to Sushiswap
✅  BCP goes to BCP page where the modal can be opened
✅  Exchange defaults to buying DOUGH 
✅  Exchange has swapped DAI for USDC + check with decimals
✅ DEFI++ added to exchange page

**Review by AdrianAddesso — APPROVED** [review]
Approved. All working as expected.

✅  PLAY goes to Sushiswap
✅  DEFI++ goes to exchange page <- It does, but the exchange does not Show DeFi++ listed when accessed from the PiePage
✅  BCP goes to BCP page where the modal can be opened
✅  Exchange defaults to buying DOUGH 
✅  Exchange has swapped DAI for USDC + check with decimals
✅ DEFI++ added to exchange page
